qualtran/bloqs/qft/qft_phase_gradient.py

```diff
@@ -54,7 +54,11 @@ class QFTPhaseGradient(GateWithRegisters):
 
     @cached_property
     def signature(self) -> 'Signature':
-        return Signature.build(q=self.bitsize, phase_grad=self.bitsize)
+        if self.bitsize == 1:
+            return Signature([Register("q", QBit()), Register("phase_grad", QBit())])
+        return Signature(
+            [Register("q", QUInt(self.bitsize)), Register("phase_grad", QUInt(self.bitsize))]
+        )
```

Comment:
Ditto: Do we need the `QUInt` vs `QBit` special casing?

Reply:
first, it would be helpful to have a `Registers:` section in the docstring so I know what the "q" register is supposed to be. 

Does it make sense to do a 1-bit QFT?

Reply:
> Does it make sense to do a 1-bit QFT?

Yes, it's a hadamard gate. And when writing generic code that works for arbitrary `bitsize`, it's useful to not have special casing checks at the call site to delegate to a hadamard or a QFT separately; when the decomposition can take care of that. Costings would be correct in both the cases.